gufe/settings/models.py

```diff
@@ -0,0 +1,143 @@
+"""
+Pydantic models used for storing settings.
+"""
+
+import abc
+from datetime import date
+from typing import Literal, Optional, Union
+
+from openff.models.models import DefaultModel
+from openff.models.types import FloatQuantity
+from pydantic import Extra, FilePath, PositiveFloat, Field
+from typing_extensions import TypedDict
+
+
+class SettingsBaseModel(DefaultModel):
+    """Settings and modifications we want for all settings classes.
+    """
+
+    class Config:
+        extra = Extra.forbid
+        # Immutability in python is never strict.
+        # If developers are determined/stupid
+        # they can always modify a so-called "immutable" object.
+        # https://pydantic-docs.helpmanual.io/usage/models/#faux-immutability
+        allow_mutation = False
+        arbitrary_types_allowed = False
+        smart_union = True
+
+
+class vdWScale(SettingsBaseModel):
+    """
+    See the `SMIRNOFF vdW specification <https://openforcefield.github.io/standards/standards/smirnoff/#vdw>`_
+    for more details.
+    """
+    scale12: float = 0.0
+    scale13: float = 0.0
+    scale14: float = 0.5
+    scale15: float = 1.0
+
+
+class ElectrostaticScale(SettingsBaseModel):
+    """
+    See the `SMIRNOFF Electrostatics specification <https://openforcefield.github.io/standards/standards/smirnoff/#electrostatics>`_
+    for more details.
+    """
+    scale12: float = 0.0
+    scale13: float = 0.0
+    scale14: float = 0.833333
+    scale15: float = 1.0
+
+
+class VdWSettings(SettingsBaseModel):
+    """Settings for van der Waals force.
+
+    See the `SMIRNOFF vdW specification <https://openforcefield.github.io/standards/standards/smirnoff/#vdw>`_
+    for more details.
+    """
+
+    combining_rules: Literal["Lorentz-Berthelot"]
```

Comment:
I realise populating all these might take time / effort, I'd be happy to see these being raised as issues for folks to pitch in a short amount of time on (rather than holding back the PR).

Reply:
That sounds good, but can you give an example of what you would like to see here? Link to the openff doc? Wikipedia page? Or just a blurb: this mixing rule means (then the math expression)